src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -0,0 +1,121 @@
+        /// <param name="armClient">arm client</param>
+        /// <param name="template">template used in deployment</param>
+        /// <param name="parameterFilePath">path to parameter file used in deployment</param>
+        /// <param name="id">id string to create the ResourceIdentifier from</param>
+        /// <param name="scope">target scope</param>
+        /// <param name="location">location to store the deployment data</param>
+        /// <returns>deployment succeeded/failed message</returns>
+        public static async Task<string> CreateDeployment(
+            IDeploymentCollectionProvider deploymentCollectionProvider,
+            ArmClient armClient,
+            string template,
+            string parameterFilePath,
+            string id,
+            string scope,
+            string location)
+        {
+            if ((scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup) &&
+                string.IsNullOrWhiteSpace(location))
+            {
+                return LangServerResources.MissingLocationDeploymentFailedMessage;
+            }
+
+            DeploymentCollection? deploymentCollection;
+            var resourceIdentifier = new ResourceIdentifier(id);
+
+            try
+            {
+                deploymentCollection = deploymentCollectionProvider.GetDeploymentCollection(armClient, resourceIdentifier, scope);
+            }
+            catch (Exception e)
+            {
+                return string.Format(LangServerResources.DeploymentFailedWithExceptionMessage, e.Message);
+            }
+
+            if (deploymentCollection is not null)
+            {
+                JsonElement parameters;
+
+                try
+                {
+                    parameters = GetParameters(parameterFilePath);
+                }
+                catch (Exception e)
+                {
+                    return e.Message;
+                }
+
+                var deploymentProperties = new DeploymentProperties(DeploymentMode.Incremental)
+                {
+                    Template = JsonDocument.Parse(template).RootElement,
+                    Parameters = parameters
+                };
+                var input = new DeploymentInput(deploymentProperties)
+                {
+                    Location = location,
+                };
+
+                string deployment = "deployment_" + DateTime.UtcNow.ToString("yyyyMMddHHmmss");
+
+                try
+                {
+                    var deploymentCreateOrUpdateAtScopeOperation = await deploymentCollection.CreateOrUpdateAsync(deployment, input);
+
+                    if (deploymentCreateOrUpdateAtScopeOperation.HasValue &&
+                        deploymentCreateOrUpdateAtScopeOperation.GetRawResponse().Status == 200)
+                    {
+                        return LangServerResources.DeploymentSucceededMessage;
```

Comment:
If it's another status result but no exception, do we need to give more info?  Are any other status values possible?

Reply:
Good question. Looks like the sdk only returns 200 or 201:
https://github.com/Azure/azure-sdk-for-net/blob/2f1462d2a9b91fb52a367feaa9ca097c922b907d/sdk/resources/Azure.ResourceManager.Resources/src/Generated/RestOperations/DeploymentsRestOperations.cs#L230

But it's not documented here:
https://docs.microsoft.com/en-us/dotnet/api/azure.resourcemanager.resources.models.deploymentcreateorupdateoperation?view=azure-dotnet-preview

Added support to return response in case of failure:
https://github.com/Azure/bicep/commit/e484bb2905d1f803873648946e2e862f4504fe8f

Thanks!